package-builder/flox-build.mk

```diff
@@ -109,7 +109,7 @@ define DEPENDS_template =
   # Compute 32-character stable string for use in stable path generation
   # based on hash of pname, current working directory and FLOX_ENV.
   $(eval $(_pvarname)_hash = $(shell ( \
-    ( echo $(_pname) $(realpath $(FLOX_ENV)) && $(_pwd) ) | $(_sha256sum) | $(_head) -c32)))
+    ( echo $(_pname) $(FLOX_ENV) && $(_pwd) ) | $(_sha256sum) | $(_head) -c32)))
```

Comment:
I'm not following your first two points but your third question is interesting. It boils down to the ability to use a "standard" flox environment for building similar projects.

For example imagine you had a bunch of projects that all built in the exact same way. Now imagine you were wanting to compare the effects of building with two different build environments. With that you could presumably `cd` into each project and run `flox build -r me/mybuildenv{1,2}` and in that case you'd want each of those invocations to generate and use a different set of temporary files. That's what I had thought, anyway ...

But for now you're right that `$(_pwd)` is likely to be unique enough. We shouldn't let this block us because it's so easily tweaked later, so happy to drop the mention of FLOX_ENV for now.